### PR TITLE
CON-140 Create contest overview screen

### PR DESCRIPTION
**Pull request by gsintrepid**
https://intrepid.atlassian.net/browse/CON-140

- Added "todos" to hook up with Abass's recycler view and styles later

@abassawo can take over this subtask to display the scoring categories.
When you're done with that I can either take it back or you can do the remaining part (display the description and the score legend [1 poor, ..., 5 excellent], which you don't create adding a category, but I think comes from the server).

### CODE REVIEW

File: app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewActivity.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+
+        setActionBarDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    public void showContestName(String contestName) {
+        setActionBarTitle(getResources().getString(R.string.status_waiting_submissions_judge_bar_title, contestName));
```
Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewActivity.java]: nice!

File: app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+
+        contestId = persistentSettings.getCurrentContestId().toString();
+        requestContestDetails();
+        requestContestStatus();
+    }
+
+    private void requestContestStatus() {
```
Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenter.java]: super nit but can you make the order of the  two "request" method definitions match the order you're calling them in.

File: app/src/main/res/layout/activity_contest_overview.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        style="@style/AppTheme.BottomNavigationButton"
+        android:layout_alignParentBottom="true"
+        android:text="@string/contest_overview_submit_button_text"
+        />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        android:paddingBottom="@dimen/activity_vertical_margin"
```
Comment (abassawo) [inline app/src/main/res/layout/activity_contest_overview.xml]: can we just make this a single android:padding attribute? I think activity_vertical_margin and horizontal have same dp value.

File: app/src/test/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        response.numSubmissionsMissing = 5;
+        when(mockRestApi.getContestStatus(any())).thenReturn(Observable.just(response));
+    }
+
+    private void setupFailedContestStatusCall() {
+        when(mockRestApi.getContestStatus(any())).thenReturn(error(new Throwable()));
+    }
+
+    @Test
+    public void onViewCreatedShouldShowContestNameWhenSuccessfullyRetrievedContactDetails() throws Exception {
```
Comment (abassawo) [inline app/src/test/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenterTest.java]: probably a typo or autocorrect. change Contact to Contest

Comment (abassawo) [inline app/src/test/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenterTest.java]: We should find a more fitting name for this test case

File: app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenter.java (hunk 2 above)
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenter.java]: makes sense!

File: app/src/main/res/layout/activity_contest_overview.xml (hunk 3 above)
Comment (gsintrepid) [inline app/src/main/res/layout/activity_contest_overview.xml]: actually they don't, `vertical` is 16dp and `horizontal` is 64dp!

File: app/src/test/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenterTest.java (hunk 4 above)
Comment (gsintrepid) [inline app/src/test/java/io/intrepid/contest/screens/contestoverview/ContestOverviewPresenterTest.java]: Good catch!

**Pull request merged**